### PR TITLE
Fix sigint again

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

File: bluesky/tests/conftest.py [hunk 1]
```diff
@@ -10,6 +10,15 @@ def fresh_RE(request):
     loop.set_debug(True)
     RE = RunEngine({}, loop=loop)
     RE.ignore_callback_exceptions = False
+
+    def clean_al():
```
Comment (danielballan) [inline bluesky/tests/conftest.py]: Is this supposed say `clean_all`? Maybe `al` stands for something that I'm not seeing; if so, I'll leave this :sheep: here preemptively.

Comment (tacaswell) [inline bluesky/tests/conftest.py]: It should be `clean_el` (for event loop).

File: bluesky/tests/test_devices.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 if ophyd:
     from ophyd import Component as Cpt, Device, Signal
+    ophyd.setup_ophyd()
```
Comment (danielballan) [inline bluesky/tests/test_devices.py]: This probably explains some of our intermittent failures. Good catch.

File: bluesky/examples.py [hunk 3]
```diff
@@ -154,8 +153,14 @@ def __getstate__(self):
         return (self.name, self._fields, self.read_attrs, self.conf_attrs,
                 self._monitor_intervals)
 
-    def trigger(self):
-        "No-op: returns a status object immediately marked 'done'."
+    def trigger(self, *, delay_time=0):
```
Comment (danielballan) [inline bluesky/examples.py]: Bad choice of words. I agree that ophyd is the reference implementation. With the recent changes, `Reader` is a useful public class in its own right, not just a base class. Is there a good reason to have it present a non-standard API? This `delay_time` seems like a init-time decision ("how should this fake device behave on trigger") and not something that should be passed through a Msg into the RE into the device at trigger-time.

Comment (tacaswell) [inline bluesky/examples.py]: to avoid picking a magic attribute name and spending too much time walking the class tree.

**Comment by tacaswell** [comment]
This should be merged and a more agressive re-working of sig-int handling will come as a follow on

File: bluesky/tests/conftest.py (hunk 1 above)
Comment (danielballan) [inline bluesky/tests/conftest.py]: Plan to fix this?

Comment (danielballan) [inline bluesky/tests/conftest.py]: Nevermind; I merged locally and added a commit.

**Pull request merged**